cli/src/commands/fix.rs

```diff
@@ -150,6 +148,26 @@ pub(crate) fn cmd_fix(
         .parse_file_patterns(ui, &args.paths)?
         .to_matcher();
 
+    fix_revisions(
+        ui,
+        &mut workspace_command,
+        &root_commits,
+        &matcher,
+        args.include_unchanged_files,
+    )?;
+    Ok(())
+}
+
+pub(crate) fn fix_revisions(
+    ui: &mut Ui,
+    workspace_command: &mut WorkspaceCommandHelper,
+    root_commits: &[CommitId],
+    matcher: &dyn Matcher,
+    include_unchanged_files: bool,
+) -> Result<HashMap<CommitId, CommitId>, CommandError> {
+    let workspace_root = workspace_command.workspace_root().to_owned();
+    let path_converter = workspace_command.path_converter().to_owned();
+    let tools_config = get_tools_config(ui, workspace_command.settings())?;
     let mut tx = workspace_command.start_transaction();
```

Comment:
Maybe it's better to have the caller start the transaction? That gives them more control over it, in case they want to do something else in the same transaction. Oh, one caller calls this function repeatedly. Maybe it actually would prefer to create just one transaction?